capstone/src/main/java/com/google/sps/Objects/User.java

```diff
@@ -171,4 +175,23 @@ public Entity toEntity() {
     userEntity.setProperty("interests", interests);
     return userEntity;
   }
+
+  /**
+   * Helper method for {@code fromEntity()}. Returns a list of badges given a list of badge ids.
+   */
+  private static LinkedHashSet<Badge> getBadgeList(LinkedHashSet<Long> badgeIds) { 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    LinkedHashSet<Badge> badges = new LinkedHashSet<>();
+    for (long badgeId : badgeIds) {
+      Key badgeKey = KeyFactory.createKey("Badge", badgeId);
+      Badge badge = null;
+      try { 
+        badge = Badge.fromEntity(datastore.get(badgeKey));
+      } catch(EntityNotFoundException e) {
+        // what to do here?
```

Comment:
are we doing the alert thing on the client side? If so we would need a field in the json that contains the exception messages

Reply:
Yup, that makes sense.

Reply:
The issue in this specific case is that since this is an object file, I'm not sure how to send the response.

Reply:
Instead of catching the error here could you instead pass it along to the servlets by throwing the error and catching the error when you call fromEntity?